modules/cmr_module/CmrApi.cc

```diff
@@ -764,42 +769,85 @@ const {
     return result;
 }
 
-
-/**
- * https://cmr.earthdata.nasa.gov/legacy-services/rest/providers/LPDAAC_ECS.json
- * @param provider_id
- * @return
- */
-Provider CmrApi::get_provider(const std::string &provider_id) const
+void CmrApi::get_providers_list(std::vector<std::string> &provider_ids) const
 {
     JsonUtils json;
 
-    string cmr_query_url = BESUtil::pathConcat(d_cmr_providers_search_endpoint_url,provider_id);
-    cmr_query_url += ".json";
-    BESDEBUG(MODULE, prolog << "CMR Providers Search Request Url: : " << cmr_query_url << endl);
+    stringstream cmr_query_url;
```

Comment:
It's hard to see from the 'diff view' in github so maybe stringstream is needed, but in general, string's + operator is a much faster way to combine strings. It might not be the fastest (append might be faster still, but with the compiler's optimizer on, it seems to be really close). There are various sources that quote measurements of 25-30 times faster when code is optimized. FWIW.